front/app/containers/Admin/pagesAndMenu/containers/GenericHeroBannerForm/BannerImageFields/index.tsx

```diff
@@ -94,37 +86,70 @@ const BannerImageField = ({
       setBannerError(formatMessage(messages.noHeader));
       return;
     }
+  }, [headerBg, formatMessage]);
 
-    setBannerError(null);
-  }, [headerBg, formatMessage, setFormStatus]);
+  const convertHeaderToUploadFile = async (
+    fileInfo: string | null | undefined
+  ) => {
+    if (fileInfo) {
+      const tenantHeaderBg = await convertUrlToUploadFile(fileInfo);
+
+      setHeaderLocalDisplayImage(
+        !isNilOrError(tenantHeaderBg) ? tenantHeaderBg : null
+      );
+    }
+  };
 
-  const handleOnAddImageToUploader = (newImage: UploadFile[]) => {
+  const handleOnAddImageToUploader = (newImages: UploadFile[]) => {
     // this base64 value is sent to the API
-    onAddImage(newImage[0].base64);
+    onAddImage(newImages[0].base64);
     // this value is used for local display
-    setHeaderLocalDisplayImage([newImage[0]]);
+    setHeaderLocalDisplayImage(newImages[0]);
+    setBannerError(null);
   };
 
   const handleOnRemoveImageFromUploader = () => {
     onRemoveImage();
     setHeaderLocalDisplayImage(null);
   };
 
-  const imageIsSaved = headerLocalDisplayImage?.[0].remote || false;
   const hasLocalHeaderImage = !isNilOrError(headerLocalDisplayImage);
+  const imageShouldBeSaved = headerLocalDisplayImage
+    ? headerLocalDisplayImage.remote
+    : false;
+
+  const showConditions = (bannerLayout: THomepageBannerLayout) => {
+    const conditions: {
+      [key in THomepageBannerLayout]: {
+        [key in TBannerLayoutComponent]: boolean;
+      };
+    } = {
+      full_width_banner_layout: {
+        image_cropper: false,
+        preview_device: hasLocalHeaderImage,
+        overlay_controls: hasLocalHeaderImage,
+      },
+      two_row_layout: {
+        image_cropper: false,
+        preview_device: hasLocalHeaderImage,
+        overlay_controls: false,
+      },
+      two_column_layout: {
+        image_cropper: false,
+        preview_device: hasLocalHeaderImage,
+        overlay_controls: false,
+      },
+      fixed_ratio_layout: {
+        image_cropper: hasLocalHeaderImage && imageShouldBeSaved,
+        preview_device: false,
+        // For the fixed_ratio_layout we only show it for a saved image.
+        // An unsaved image should show the image cropper instead.
+        overlay_controls: hasLocalHeaderImage && imageShouldBeSaved,
+      },
+    };
 
-  const displayImageCropper =
-    hasLocalHeaderImage &&
-    !imageIsSaved &&
-    bannerLayout === 'fixed_ratio_layout';
-
-  const displayPreviewDevice =
-    hasLocalHeaderImage && bannerLayout !== 'fixed_ratio_layout';
-
-  const displayOverlayControls =
-    hasLocalHeaderImage &&
-    (bannerLayout === 'full_width_banner_layout' ||
-      (bannerLayout === 'fixed_ratio_layout' && imageIsSaved));
+    return conditions[bannerLayout];
+  };
```

Comment:
Imo much clearer than the display conditions. Now we need to define it explicitly for each layout. Maybe this is almost as good as the unit/integration tests and could almost replace them. The unit/integration tests act as documentation/reference though.